crates/bevy_transform/src/systems.rs

```diff
@@ -66,21 +65,12 @@ fn propagate_recursive(
         (&Transform, Changed<Transform>, &mut GlobalTransform),
         With<Parent>,
     >,
-    parent_query: &Query<&Parent>,
-    children_query: &Query<(&Children, Changed<Children>), (With<Parent>, With<GlobalTransform>)>,
-    expected_parent: Entity,
+    parent_query: &Query<(Entity, &Parent, Changed<Parent>)>,
+    children_query: &Query<&Children, (With<Parent>, With<GlobalTransform>)>,
     entity: Entity,
     mut changed: bool,
     // We use a result here to use the `?` operator. Ideally we'd use a try block instead
 ) {
-    let Ok(actual_parent) = parent_query.get(entity) else {
-        panic!("Propagated child for {:?} has no Parent component!", entity);
-    };
-    assert_eq!(
-        actual_parent.get(), expected_parent,
-        "Malformed hierarchy. This probably means that your hierarchy has been improperly maintained, or contains a cycle"
-    );
-
     let global_matrix = {
         let Ok((transform, transform_changed, mut global_transform)) =
             // SAFETY: This call cannot create aliased mutable references.
```

Comment:
This safety comment should be updated since it says "The above assertion ensures that ...", but the assertion is no longer there.

Reply:
This looks solved to me, but I can't mark it as such.